### PR TITLE
Make handleSummary func timeout configurable

### LINKED ISSUE #3976: Ability to configure the 120 second timeout when generating end of test statistics
### Feature Description

Hello,

At the moment, when we run longer tests (4-5 hours+), the end of test summary generation is failing with
`"handleSummary() execution timed out after 120 seconds"`

(tracking workarounds as part of the post on https://community.grafana.com/t/execution-timed-out-after-120-seconds/132323)

Would it be possible to add a way to configure this timeout, if we're running longer tests we don't really mind if it takes 10-20 minutes to process the stats at the end?

Thanks.

### Suggested Solution (optional)

_No response_

### Already existing or connected issues / PRs (optional)

_No response_

### PR DESCRIPTION
**Pull request by alex-kattathra-johnson**
## What?

It makes the timeout for the `handleSummary` function configurable

## Why?

To solve issue #3976. To allow for generating summary for long running tests

## Checklist

- [x] I have performed a self-review of my code.
- [x] I have added tests for my changes.
- [x] I have run linter locally (`make lint`) and all checks pass.
- [x] I have run tests locally (`make tests`) and all tests pass.
- [x] I have commented on my code, particularly in hard-to-understand areas.

## Related PR(s)/Issue(s)
Closes #3976

### CODE REVIEW

**Comment by taylorflatt** [comment]
I'd make sure to update the docs as well if this change does move forward: https://grafana.com/docs/k6/latest/using-k6/k6-options/reference/

See: https://github.com/grafana/k6-docs

File: js/summary_test.go [hunk 1]
```diff
@@ -55,7 +55,10 @@ func TestTextSummary(t *testing.T) {
 			runner, err := getSimpleRunner(
 				t, "/script.js",
 				fmt.Sprintf(`
-					exports.options = {summaryTrendStats: %s};
+					exports.options = {
+						summaryTrendStats: %s,
+						handleSummaryTimeout: '120s',
+					};
```
Comment (mstoykov) [inline js/summary_test.go:61]: Do you really need to update all of those tests ?

**Review by mstoykov — CHANGES_REQUESTED** [review]
Thank you for the PR :bow: 

It seems to me like you are changing way too many tests and I expect there might be some problems with the propagation of the default.

Can you try to rever each of the test changes and see if you can reduce them? I haven't commented in each place.

It also will be nice to get a test case in https://github.com/grafana/k6/blob/master/cmd/config_consolidation_test.go#L146 as that is where consolidation is usually tested as well as defaults.

File: js/summary_test.go (hunk 1 above)
Comment (alex-kattathra-johnson) [reply js/summary_test.go:61]: I think the defaults are set at the `cmd` level. But, these unit tests are testing the functionality of these respective modules independent of `cmd` execution.

Comment (alex-kattathra-johnson) [reply js/summary_test.go:61]: @mstoykov : Friendly ping on this! 😄

**Pull request closed without merging**